src/internal/naming/name-service.ts

```diff
@@ -0,0 +1,79 @@
+import { Connection, PublicKey } from '@solana/web3.js';
+import { FavouriteDomain, NAME_OFFERS_ID } from '@bonfida/name-offers';
+import {
+  NAME_PROGRAM_ID,
+  performReverseLookup,
+} from '@bonfida/spl-name-service';
+import { tryGetName as tryGetTwitterHandle } from '@cardinal/namespaces';
+import type { NameService } from 'naming/naming.interface';
+
+export class NameServiceResolver implements NameService {
+  private connection: Connection;
+  private CARDINAL_CONST_PK = 'nameXpT2PwZ2iA6DTNYTotTmiMYusBCYqwBLN2QgF4w';
+
+  constructor(readonly solanaRpcUrl: string) {
+    this.connection = new Connection(solanaRpcUrl);
```

Comment:
I guess connection could be reused from the sdk?

Reply:
Let me double check, but I didn't find connection itself...